src/useMap.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react'
+
+import { MapkitContext } from './MapkitProvider'
+import { MapOptions, propsToMapConstructionOptions } from './utils'
+
+import {
+  NumberTuple,
+  Rect,
+  RegionType,
+  createCoordinate,
+  createMapRect,
+  createCoordinateRegionFromValues,
+} from './utils'
+
+export const useMap = (defaultOptions: MapOptions = {}) => {
+  const [defaultMapOptions] = React.useState(defaultOptions)
+  let { mapkit } = React.useContext(MapkitContext)
+  let mapRef = React.useRef<HTMLDivElement>(null)
+  let [map, setMap] = React.useState<mapkit.Map>()
+
+  React.useEffect(() => {
+    if (mapkit && mapRef.current) {
+      const newMap = new mapkit.Map(
+        mapRef.current,
+        propsToMapConstructionOptions(defaultMapOptions),
+      )
+      setMap(newMap)
+    }
+  }, [mapRef, mapkit])
+
+  // Clean up the map on unmount
+  React.useEffect(() => {
+    return () => {
+      if (map) {
+        map.destroy()
+      }
+    }
+  }, [])
+
+  return {
+    mapkit,
+    map,
+    mapProps: {
+      mapkit,
+      map,
+      mapRef,
+    },
+    setRotation: React.useCallback(
```

Comment:
Why is `useCallback` needed?

Reply:
> React memoizes the callback function for you and the callback is not created multiple times.
https://stackoverflow.com/a/53177820/4076109